core/src/main/java/org/apache/brooklyn/core/entity/Entities.java

```diff
@@ -948,10 +955,22 @@ public static ManagementContext startManagement(Application app, ManagementConte
             }
         }
 
+        premanageRecursively((ManagementContextInternal)mgmt, app);
+
         mgmt.getEntityManager().manage(app);
         return mgmt;
     }
 
+    private static void premanageRecursively(ManagementContextInternal mgmt, Entity e) {
+        if (Entities.isUnmanagingOrNoLongerManaged(e) || Entities.isManagedActive(e)) {
+            log.warn("Skipping premanagement for "+e+", as it has some form of known management");
+            // skip for active and unmanaged entities
+            return;
+        }
+        mgmt.prePreManage(e);
```

Comment:
Can it be that this condition is required for a particular entity, but not for its children?

Reply:
In other words: should we proceed recursively with children?

Reply:
this method is called `premanageRecursively` so i think the answer is yes.  this is only needed for legacy code (tests), which does want the recursive behaviour.